pkg/sessions/dynamodb/dynamodb_store.go

```diff
@@ -0,0 +1,139 @@
+package dynamodb
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/aws/session"
+	dynamo "github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/apis/options"
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/apis/sessions"
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/sessions/persistence"
+)
+
+const SessionKeyName = "SessionKey"
+
+// GetPutDeleteWithContext is a subset of the dynamodbiface.DynamoDBAPI interface that defines only the operations
+// we need. This is useful for testing as it enables us to provide a custom client that doesn't have to implement the full interface.
+type GetPutDeleteWithContext interface {
+	GetItemWithContext(aws.Context, *dynamo.GetItemInput, ...request.Option) (*dynamo.GetItemOutput, error)
+	PutItemWithContext(aws.Context, *dynamo.PutItemInput, ...request.Option) (*dynamo.PutItemOutput, error)
+	DeleteItemWithContext(aws.Context, *dynamo.DeleteItemInput, ...request.Option) (*dynamo.DeleteItemOutput, error)
+}
+
+// SessionStore is an implementation of the persistence.Store
+// interface that stores sessions in dynamoDB
+type SessionStore struct {
+	dynamoService GetPutDeleteWithContext
+	tableName     string
+}
+
+type DynamoSessionItem struct {
+	SessionKey string
+	Value      []byte
+	Expiry     string
+}
+
+var now = time.Now
```

Comment:
I think you can remove this -- you only use `now()` in one spot below, in the other you are still using `time.Now()` anyway.

Reply:
Oh duh -- is this to help stub `time.Now` in future `FastForward` supporting test cases? In that case, carry on

![Homer Grass](https://media.giphy.com/media/jUwpNzg9IcyrK/giphy.gif)